main.go

```diff
@@ -1,20 +1,54 @@
 package main
 
 import (
+	"context"
 	"io/ioutil"
 	"log"
 	"net/http"
 	"os"
 	"time"
 
 	"github.com/go-git/go-git/v5"
+	"github.com/google/go-github/v47/github"
 	"github.com/julienschmidt/httprouter"
 )
 
 const repoURL = "https://github.com/RedHatInsights/insights-core"
 
 var repoPath = ""
 
+var forks_cache = make([]map[string]string, 0)
+
+func getGithubForks() {
```

Comment:
For an internally called function like this, it's a better convention to let it return an error it receives from any functions it calls, and let its caller deal with checking for whether it returns an error or not.

So in this case, the function signature would change to:

```go
func getGithubForks() error {
```

And anywhere in the function body where you call `log.Fatal(err)`, replace it with `return err`. Then in the locations where you call `getGithubForks`, you can check for an error:

```go
if err := getGithubForks(); err != nil {
    log.Fatalf("error creating github forks cache: %v", err)
}